src/redis/cache_impl.go

```diff
@@ -178,13 +167,20 @@ func (this *rateLimitCacheImpl) DoLimit(
 			expirationSeconds += this.jitterRand.Int63n(this.expirationJitterMaxSeconds)
 		}
 
+		result := &results[i]
+		key := cacheKey.key
 		// Use the perSecondConn if it is not nil and the cacheKey represents a per second Limit.
-		if perSecondConn != nil && cacheKey.perSecond {
-			pipelineAppend(perSecondConn, cacheKey.key, hitsAddend, expirationSeconds)
+		if this.perSecondClient != nil && cacheKey.perSecond {
+			if err := pipelineAppend(this.perSecondClient, key, hitsAddend, result, expirationSeconds); err != nil {
+				lastErr = err
+			}
 		} else {
-			pipelineAppend(conn, cacheKey.key, hitsAddend, expirationSeconds)
+			if err := pipelineAppend(this.client, key, hitsAddend, result, expirationSeconds); err != nil {
+				lastErr = err
+			}
 		}
 	}
+	checkError(lastErr)
```

Comment:
can you let us know why only to panic in last error, instead of panic in the first error?

Reply:
When wrote this code, I was thinking to execute all commands  and recording all errors. For simplicity, only lastErr was checked. But now thinking again, it seems check first error is enough. I will update this latter.